optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintSession.java

```diff
@@ -48,34 +56,80 @@
     private final ScoreInliner<Score_> scoreInliner;
 
     private final Map<Class<?>, BavetFromUniNode<Object>> declaredClassToNodeMap;
-    private final int nodeOrderSize;
+    private final int nodeIndexSize;
     private final Map<String, BavetScoringNode> constraintIdToScoringNodeMap;
 
     private final Map<Class<?>, List<BavetFromUniNode<Object>>> effectiveClassToNodeListMap;
 
-    private final List<Queue<BavetAbstractTuple>> nodeOrderedQueueList;
+    private final List<BavetAbstractNode> nodeIndexToNodeMap;
+    private final List<Queue<BavetAbstractTuple>> nodeIndexToDirtyTupleQueueMap;
     private final Map<Object, List<BavetFromUniTuple<Object>>> fromTupleListMap;
 
     public BavetConstraintSession(boolean constraintMatchEnabled, ScoreDefinition<Score_> scoreDefinition,
             Map<BavetConstraint<Solution_>, Score_> constraintToWeightMap) {
         this.constraintMatchEnabled = constraintMatchEnabled;
-        this.zeroScore = scoreDefinition.getZeroScore();
-        this.scoreInliner = scoreDefinition.buildScoreInliner(constraintMatchEnabled);
+        zeroScore = scoreDefinition.getZeroScore();
+        scoreInliner = scoreDefinition.buildScoreInliner(constraintMatchEnabled);
         declaredClassToNodeMap = new HashMap<>(50);
         BavetNodeBuildPolicy<Solution_> buildPolicy = new BavetNodeBuildPolicy<>(this, constraintToWeightMap.size());
         constraintToWeightMap.forEach((constraint, constraintWeight) -> {
             constraint.createNodes(buildPolicy, declaredClassToNodeMap, constraintWeight);
         });
-        this.nodeOrderSize = buildPolicy.getNodeOrderMaximum() + 1;
+        nodeIndexSize = buildPolicy.getNodeIndexMaximum() + 1;
         constraintIdToScoringNodeMap = buildPolicy.getConstraintIdToScoringNodeMap();
         effectiveClassToNodeListMap = new HashMap<>(declaredClassToNodeMap.size());
-        nodeOrderedQueueList = new ArrayList<>(nodeOrderSize);
-        for (int i = 0; i < nodeOrderSize; i++) {
-            nodeOrderedQueueList.add(new ArrayDeque<>(1000));
+        nodeIndexToNodeMap = new ArrayList<>(this.nodeIndexSize);
```

Comment:
This is not actually in use yet. The change to completely remove `Node` reference from `Tuple` seemed too invasive for this PR.

Reply:
Yes, thank you